conans/client/source.py

```diff
@@ -54,16 +53,17 @@ def is_excluded(origin_path):
             return True
         return False
 
-    linked_folders = []
     for src_dir, dirs, files in walk(src, followlinks=True):
 
         if is_excluded(src_dir):
             dirs[:] = []
             continue
 
         if os.path.islink(src_dir):
-            rel_link = os.path.relpath(src_dir, src)
-            linked_folders.append(rel_link)
+            linkto = os.readlink(src_dir)
+            dst_dir = os.path.normpath(os.path.join(dst, os.path.relpath(src_dir, src)))
+            os.symlink(linkto, dst_dir)
```

Comment:
I think there is a context in which this will be another regression. For SCM, it is being executed:
```
merge_directories(local_sources_path, dest_dir, excluded=excluded)
```
copying things from ``local_sources_path``. If they have symlinks, they will be pointing to the user folder, not converted to relative.

Reply:
I have submitted two changes, to see what happens with CI:
- if symlink is absolute, convert to relative
- if symlink is pointing outside the source, skip it

This is more or less what the "FileCopier.link_folder" does.